avm/res/db-for-postgre-sql/flexible-server/tests/e2e/defaults/main.test.bicep

```diff
@@ -40,16 +40,25 @@ resource resourceGroup 'Microsoft.Resources/resourceGroups@2023-07-01' = {
 // ============== //
 
 @batchSize(1)
-module testDeployment '../../../main.bicep' = [for iteration in [ 'init', 'idem' ]: {
-  scope: resourceGroup
-  name: '${uniqueString(deployment().name, resourceLocation)}-test-${serviceShort}-${iteration}'
-  params: {
-    name: '${namePrefix}${serviceShort}001'
-    location: resourceLocation
-    administratorLogin: 'adminUserName'
-    administratorLoginPassword: password
-    skuName: 'Standard_B2s'
-    tier: 'Burstable'
-    geoRedundantBackup: 'Enabled'
+module testDeployment '../../../main.bicep' = [
```

Comment:
Hey @donheerschap,
please note that the `defaults` test (as per its description) should only contain the required parameters. However, you happened to have added a few optional ones and hence this naturally raises the question as to why 😉

Reply:
Because the PSrule validation forced me so (even on the main branch) and I couldn't figure out how to exclude those. Could you suggest how? When I would go back again to only defaults it would trigger those again.

Reply:
Hey @donheerschap,
let me take a closer look at get back to this :) It's like you said, that the defaults should be waf-aligned and anything that requires a user input should be excluded via setting from the default test and only be set on the `waf-aligned` one. At the same time, only the reliability tests should matter, while the rest of the PSRules 'should' be satisfied but are not enforced. Like I said, I'll have a closer look once I can and will recommend what to do :)

Reply:
Hey @donheerschap,
I wasn't able to find any instances where PSRule failed your pipelines (at least in the last 3 weeks). If you have a run handy, that would be neat 😄

At least regarding the defaults test I'd recommend to remove anything non-required that was added to keep it on point and not add to the confusion. PSRule may show warnings, but it should not fail the pipeline.
In that state we can merge the PR then and it will run all the way and publish the module.

Now, to tackle PSRule in the correct way a few things should be done (under consideration of common sense)
- Defaults in the module should be set to WAF/PSRule-aligned values. This only works for values that don't require user input. These can be set in the waf-aligned test then
- To then tell PSRule to not complain about those parameters, they can be excluded in the [min-suppress.Rule.yaml](https://github.com/Azure/bicep-registry-modules/blob/main/avm/utilities/pipelines/staticValidation/psrule/.ps-rule/min-suppress.Rule.yaml) file (accompanied by a comment as to why). A case of common sense is for example a rule like `Azure.Automation.ManagedIdentity` which in theory makes sense (i.e., use MSIs instead of anymous access keys), but not using either a key or MSI should be just as good / not every service needs an identity

Reply:
Once we got this tackled we only need @josearper to give us his blessing 😉

Reply:
I think I will have some time available next Monday to continue :).